src/Interpreters/ActionsVisitor.h

```diff
@@ -120,6 +121,12 @@ class ActionsMatcher
 public:
     using Visitor = ConstInDepthNodeVisitor<ActionsMatcher, true>;
 
+    enum class WindowDependancyState
```

Comment:
Why not just `bool depends_on_window` ?

Reply:
It's not affecting performance in any way, but IMHO makes code more readable. This variable describes the current state of the visitor, so enum represents it more straightforwardly than boolean.